src/pylibsshext/channel.pyx

```diff
@@ -61,8 +61,8 @@ cdef class Channel:
             raise LibsshChannelException("Failed to open_session: [%d]" % rc)
 
     def __dealloc__(self):
-        libssh.ssh_channel_free(self._libssh_channel)
-        self._libssh_channel = NULL
+        if self._libssh_channel is not NULL:
```

Comment:
You already perform the `NULL` check in `close()` so the check here is redundant.